File: src/main/java/fr/herobrine/autospeller/client/integration/AutospellerModMenuIntegration.java
package fr.herobrine.autospeller.client.integration;

//? fabric {
import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;

public class AutospellerModMenuIntegration implements ModMenuApi {

	@Override
	public ConfigScreenFactory<?> getModConfigScreenFactory() {
		return AutospellerModMenuKtIntegration.INSTANCE.getModConfigScreenFactory();
	}
}
//?}

File: src/client/kotlin/fr/herobrine/autospeller/client/integration/AutospellerModMenuIntegration.kt
package fr.herobrine.autospeller.client.integration

import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi
import fr.herobrine.autospeller.client.config.AutospellerConfiguration
import fr.herobrine.autospeller.client.config.MOD_CONFIG_HANDLER

object AutospellerModMenuIntegration: ModMenuApi {

    override fun getModConfigScreenFactory(): ConfigScreenFactory<*> {
        return { parent ->
            MOD_CONFIG_HANDLER.instance().generateScreen(parent)
        }
    }
}